### ISSUE TITLE
how to install on opam 2?

### ISSUE THREAD

**Issue by houndofsound** [ocamllabs/multicore-opam#11]
hi
tried the given instructions and says no compiler of that name (due to only looking in the default repo).
so tried this:
$ opam switch create 4.06.1+multicore --repositories=multicore
[ERROR] Could not resolve set of base packages:
        The following dependencies couldn't be met:
          - ocaml-variants → base-threads
            unknown package

how do I get the depended upon packages from one repo while getting the modded compiler from another?
ta.

**Comment by ChrisVine** [comment]
I came across exactly the same problem.  I think the issue is that with opam-2.0, repositories are associated with switches rather than being globally available, and when you add the remote repository for multicore-ocaml, there is no switch for multicore existing for the repository to attach to.

This is going to make it difficult for any non-default opam repository to offer a custom ocaml compiler.  I got around this by first creating an empty switch for multicore, and then adding the repository to it.  This worked for me:

opam switch create 4.06.1+multicore --empty
opam switch 4.06.1+multicore
opam repository add multicore https://github.com/ocamllabs/multicore-opam.git
opam update
opam install ocaml-variants.4.06.1+multicore

It would be nice if the documentation could be updated for opam-2.0.
